File: projetos-fenix-para-estudo/t2ti-erp-fenix-menu/fontes/frontend/mvc/lib/src/model/pessoa_endereco.dart
class PessoaEndereco implements Comparable<PessoaEndereco> {
  PessoaEndereco();

  String 
        _id,
        _idPessoa,
        _logradouro,
        _numero,
        _bairro,
        _municipioIbge,
        _uf,
        _cep,
        _cidade,
        _complemento,
        _principal,
        _entrega,
        _cobranca,
        _correspondencia;

  String get id => _id;
  set id(String value){
    if (value == null) value = "";
    _id = value;
  }

  String get idPessoa => _idPessoa;
  set idPessoa(String value){
    if (value == null) value = "";
    _idPessoa = value;
  }

  String get logradouro => _logradouro;
  set logradouro(String value){
    if (value == null) value = "";
    _logradouro = value;
  }

  String get numero => _numero;
  set numero(String value){
    if (value == null) value = "";
    _numero = value;
  }

  String get bairro => _bairro;
  set bairro(String value){
    if (value == null) value = "";
    _bairro = value;
  }

  String get municipioIbge => _municipioIbge;
  set municipioIbge(String value){
    if (value == null) value = "";
    _municipioIbge = value;
  }

  String get uf => _uf;
  set uf(String value){
    if (value == null) value = "";
    _uf = value;
  }

  String get cep => _cep;
  set cep(String value){
    if (value == null) value = "";
    _cep = value;
  }

  String get cidade => _cidade;
  set cidade(String value){
    if (value == null) value = "";
    _cidade = value;
  }

  String get complemento => _complemento;
  set complemento(String value){
    if (value == null) value = "";
    _complemento = value;
  }

  String get principal => _principal;
  set principal(String value){
    if (value == null) value = "";
    _principal = value;
  }

  String get entrega => _entrega;
  set entrega(String value){
    if (value == null) value = "";
    _entrega = value;
  }

  String get cobranca => _cobranca;
  set cobranca(String value){
    if (value == null) value = "";
    _cobranca = value;
  }

  String get correspondencia => _correspondencia;
  set correspondencia(String value){
    if (value == null) value = "";
    _correspondencia = value;
  }

  PessoaEndereco.fromMap(Map m) {
    id = m["id"];
    idPessoa = m["idPessoa"];
    logradouro = m["logradouro"];
    numero = m["numero"];
    bairro = m["bairro"];
    municipioIbge = m["municipioIbge"];
    uf = m["uf"];
    cep = m["cep"];
    cidade = m["cidade"];
    complemento = m["complemento"];
    principal = m["principal"];
    entrega = m["entrega"];
    cobranca = m["cobranca"];
    correspondencia = m["correspondencia"];
  }

  Map<String, dynamic> get toMap {
    return {
      "id":_id,
      "idPessoa":_idPessoa,
      "logradouro":_logradouro,
      "numero":_numero,
      "bairro":_bairro,
      "municipioIbge":_municipioIbge,
      "uf":_uf,
      "cep":_cep,
      "cidade":_cidade,
      "complemento":_complemento,
      "principal":_principal,
      "entrega":_entrega,
      "cobranca":_cobranca,
      "correspondencia":_correspondencia,
    };
  }  

  @override
  int compareTo(PessoaEndereco other) => _logradouro.compareTo(other._logradouro);
}

File: projetos-fenix-para-estudo/t2ti-erp-fenix-cadastros-completo/fontes/backend/delphi/dmvc/Model/Cadastros/PessoaEndereco.pas
{*******************************************************************************
Title: T2Ti ERP Fenix                                                                
Description: Model relacionado à tabela [PESSOA_ENDERECO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2020 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************}
unit PessoaEndereco;

interface

uses
  MVCFramework.Serializer.Commons, ModelBase;

type

  [MVCNameCase(ncLowerCase)]
  TPessoaEndereco = class(TModelBase)
  private
    FId: Integer;
    FIdPessoa: Integer;
    FLogradouro: string;
    FNumero: string;
    FBairro: string;
    FMunicipioIbge: Integer;
    FUf: string;
    FCep: string;
    FCidade: string;
    FComplemento: string;
    FPrincipal: string;
    FEntrega: string;
    FCobranca: string;
    FCorrespondencia: string;
      
  public
    procedure ValidarInsercao; override;
    procedure ValidarAlteracao; override;
    procedure ValidarExclusao; override;

    [MVCColumnAttribute('ID', True)]
	[MVCNameAsAttribute('id')]
	property Id: Integer read FId write FId;
    [MVCColumnAttribute('ID_PESSOA')]
	[MVCNameAsAttribute('idPessoa')]
	property IdPessoa: Integer read FIdPessoa write FIdPessoa;
    [MVCColumnAttribute('LOGRADOURO')]
	[MVCNameAsAttribute('logradouro')]
	property Logradouro: string read FLogradouro write FLogradouro;
    [MVCColumnAttribute('NUMERO')]
	[MVCNameAsAttribute('numero')]
	property Numero: string read FNumero write FNumero;
    [MVCColumnAttribute('BAIRRO')]
	[MVCNameAsAttribute('bairro')]
	property Bairro: string read FBairro write FBairro;
    [MVCColumnAttribute('MUNICIPIO_IBGE')]
	[MVCNameAsAttribute('municipioIbge')]
	property MunicipioIbge: Integer read FMunicipioIbge write FMunicipioIbge;
    [MVCColumnAttribute('UF')]
	[MVCNameAsAttribute('uf')]
	property Uf: string read FUf write FUf;
    [MVCColumnAttribute('CEP')]
	[MVCNameAsAttribute('cep')]
	property Cep: string read FCep write FCep;
    [MVCColumnAttribute('CIDADE')]
	[MVCNameAsAttribute('cidade')]
	property Cidade: string read FCidade write FCidade;
    [MVCColumnAttribute('COMPLEMENTO')]
	[MVCNameAsAttribute('complemento')]
	property Complemento: string read FComplemento write FComplemento;
    [MVCColumnAttribute('PRINCIPAL')]
	[MVCNameAsAttribute('principal')]
	property Principal: string read FPrincipal write FPrincipal;
    [MVCColumnAttribute('ENTREGA')]
	[MVCNameAsAttribute('entrega')]
	property Entrega: string read FEntrega write FEntrega;
    [MVCColumnAttribute('COBRANCA')]
	[MVCNameAsAttribute('cobranca')]
	property Cobranca: string read FCobranca write FCobranca;
    [MVCColumnAttribute('CORRESPONDENCIA')]
	[MVCNameAsAttribute('correspondencia')]
	property Correspondencia: string read FCorrespondencia write FCorrespondencia;
      
  end;

implementation

{ TPessoaEndereco }



procedure TPessoaEndereco.ValidarInsercao;
begin
  inherited;

end;

procedure TPessoaEndereco.ValidarAlteracao;
begin
  inherited;

end;

procedure TPessoaEndereco.ValidarExclusao;
begin
  inherited;

end;

end.